File: app/src/main/java/ru/kuchanov/scpquiz/ui/dialog/CC3LicenseDialogFragment.java
package ru.kuchanov.scpquiz.ui.dialog;

import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.text.Html;
import android.text.method.LinkMovementMethod;
import android.view.View;
import android.widget.TextView;

import com.afollestad.materialdialogs.MaterialDialog;

import java.util.Locale;

import javax.inject.Inject;

import ru.kuchanov.scpquiz.R;
import ru.kuchanov.scpquiz.controller.manager.preference.MyPreferenceManager;
import ru.kuchanov.scpquiz.di.Di;
import toothpick.Toothpick;

public class CC3LicenseDialogFragment extends DialogFragment {

    public static final String TAG = CC3LicenseDialogFragment.class.getSimpleName();

    @Inject
    MyPreferenceManager mMyPreferenceManager;

    public static DialogFragment newInstance() {
        return new CC3LicenseDialogFragment();
    }

    @Override
    public void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Toothpick.inject(this, Toothpick.openScope(Di.Scope.APP));
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(final Bundle savedInstanceState) {
        //icons from https://github.com/hjnilsson/country-flags/tree/master/svg

        final MaterialDialog.Builder dialogTextSizeBuilder = new MaterialDialog.Builder(getActivity());
        dialogTextSizeBuilder
                .customView(R.layout.dialog_cc3_license, false)
                .title(R.string.attention)
                .cancelable(false)
                .positiveText(R.string.i_accept)
                .negativeText(android.R.string.no)
                .onPositive((dialog1, which) -> {
                    mMyPreferenceManager.setPersonalDataAccepted(true);
                    dismissAllowingStateLoss();
                })
                .onNegative((dialog1, which) -> {
                    mMyPreferenceManager.setPersonalDataAccepted(false);
                    getActivity().finish();
                });

        final MaterialDialog dialog = dialogTextSizeBuilder.build();

        if (dialog.getCustomView() != null) {
            final View en = dialog.getCustomView().findViewById(R.id.en);
            final View ru = dialog.getCustomView().findViewById(R.id.ru);
            final View pl = dialog.getCustomView().findViewById(R.id.pl);
            final View de = dialog.getCustomView().findViewById(R.id.de);
            final View fr = dialog.getCustomView().findViewById(R.id.fr);
            final View es = dialog.getCustomView().findViewById(R.id.es);
            final View it = dialog.getCustomView().findViewById(R.id.it);
            final View pt = dialog.getCustomView().findViewById(R.id.pt);
            final View ch = dialog.getCustomView().findViewById(R.id.ch);

            final TextView content = dialog.getCustomView().findViewById(R.id.content);
            content.setMovementMethod(LinkMovementMethod.getInstance());

            final String siteUrl = "http://www.scp-wiki.net/";

            final String langCode = Locale.getDefault().getLanguage();

            switch (langCode) {
                case "ru":
                    setContentText(content, getString(R.string.license_ru, siteUrl, siteUrl));
                    break;
                case "en":
                    setContentText(content, getString(R.string.license_en, siteUrl, siteUrl));
                    break;
                case "pl":
                    setContentText(content, getString(R.string.license_pl, siteUrl, siteUrl));
                    break;
                case "de":
                    setContentText(content, getString(R.string.license_de, siteUrl, siteUrl));
                    break;
                case "fr":
                    setContentText(content, getString(R.string.license_fr, siteUrl, siteUrl));
                    break;
                case "es":
                    setContentText(content, getString(R.string.license_es, siteUrl, siteUrl));
                    break;
                case "it":
                    setContentText(content, getString(R.string.license_it, siteUrl, siteUrl));
                    break;
                case "pt":
                    setContentText(content, getString(R.string.license_pt, siteUrl, siteUrl));
                    break;
                case "ch":
                    setContentText(content, getString(R.string.license_ch, siteUrl, siteUrl));
                    break;
                default:
                    throw new IllegalArgumentException("unexpected lang: " + langCode);
            }

            en.setOnClickListener(view -> setContentText(content, getString(R.string.license_en, siteUrl, siteUrl)));
            pl.setOnClickListener(view -> setContentText(content, getString(R.string.license_pl, siteUrl, siteUrl)));
            ru.setOnClickListener(view -> setContentText(content, getString(R.string.license_ru, siteUrl, siteUrl)));
            de.setOnClickListener(view -> setContentText(content, getString(R.string.license_de, siteUrl, siteUrl)));
            fr.setOnClickListener(view -> setContentText(content, getString(R.string.license_fr, siteUrl, siteUrl)));
            es.setOnClickListener(view -> setContentText(content, getString(R.string.license_es, siteUrl, siteUrl)));
            it.setOnClickListener(view -> setContentText(content, getString(R.string.license_it, siteUrl, siteUrl)));
            pt.setOnClickListener(view -> setContentText(content, getString(R.string.license_pt, siteUrl, siteUrl)));
            ch.setOnClickListener(view -> setContentText(content, getString(R.string.license_ch, siteUrl, siteUrl)));
        }
        return dialog;
    }

    private static void setContentText(final TextView content, final String text) {
        content.setText(Html.fromHtml(text));
    }
}

File: app/src/main/java/com/scp/scpexam/ui/dialog/CC3LicenseDialogFragment.kt
package com.scp.scpexam.ui.dialog

import android.app.Dialog
import android.os.Bundle
import android.text.Html
import android.text.method.LinkMovementMethod
import android.view.View
import android.widget.TextView
import com.afollestad.materialdialogs.MaterialDialog
import com.scp.scpexam.R
import com.scp.scpexam.controller.manager.preference.MyPreferenceManager
import com.scp.scpexam.di.Di
import toothpick.Toothpick
import java.util.*
import javax.inject.Inject

class CC3LicenseDialogFragment : androidx.fragment.app.DialogFragment() {

    @Inject
    lateinit var mMyPreferenceManager: MyPreferenceManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Toothpick.inject(this, Toothpick.openScope(Di.Scope.APP))
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        //icons from https://github.com/hjnilsson/country-flags/tree/master/svg

        val dialogTextSizeBuilder = MaterialDialog.Builder(activity!!)
        dialogTextSizeBuilder
                .customView(R.layout.dialog_cc3_license, false)
                .title(R.string.attention)
                .cancelable(false)
                .positiveText(R.string.i_accept)
                .negativeText(android.R.string.no)
                .onPositive { _, _ ->
                    mMyPreferenceManager.setPersonalDataAccepted(true)
                    dismissAllowingStateLoss()
                }
                .onNegative { _, _ ->
                    mMyPreferenceManager.setPersonalDataAccepted(false)
                    activity!!.finish()
                }

        val dialog = dialogTextSizeBuilder.build()

        dialog.customView?.let { customView ->
            val en = customView.findViewById<View>(R.id.en)
            val ru = customView.findViewById<View>(R.id.ru)
            val pl = customView.findViewById<View>(R.id.pl)
            val de = customView.findViewById<View>(R.id.de)
            val fr = customView.findViewById<View>(R.id.fr)
            val es = customView.findViewById<View>(R.id.es)
            val it = customView.findViewById<View>(R.id.it)
            val pt = customView.findViewById<View>(R.id.pt)
            val ch = customView.findViewById<View>(R.id.ch)

            val content = customView.findViewById<TextView>(R.id.content)
            content.movementMethod = LinkMovementMethod.getInstance()

            val siteUrl = "http://www.scp-wiki.net/"

            val langCode = Locale.getDefault().language

            when (langCode) {
                "ru" -> setContentText(
                        content,
                        getString(R.string.license_ru, siteUrl, siteUrl)
                )
                "en" -> setContentText(
                        content,
                        getString(R.string.license_en, siteUrl, siteUrl)
                )
                "pl" -> setContentText(
                        content,
                        getString(R.string.license_pl, siteUrl, siteUrl)
                )
                "de" -> setContentText(
                        content,
                        getString(R.string.license_de, siteUrl, siteUrl)
                )
                "fr" -> setContentText(
                        content,
                        getString(R.string.license_fr, siteUrl, siteUrl)
                )
                "es" -> setContentText(
                        content,
                        getString(R.string.license_es, siteUrl, siteUrl)
                )
                "it" -> setContentText(
                        content,
                        getString(R.string.license_it, siteUrl, siteUrl)
                )
                "pt" -> setContentText(
                        content,
                        getString(R.string.license_pt, siteUrl, siteUrl)
                )
                "ch" -> setContentText(
                        content,
                        getString(R.string.license_ch, siteUrl, siteUrl)
                )
                else -> setContentText(
                        content,
                        getString(R.string.license_en, siteUrl, siteUrl)
                )
            }

            en.setOnClickListener {
                setContentText(
                        content,
                        getString(R.string.license_en, siteUrl, siteUrl)
                )
            }
            pl.setOnClickListener {
                setContentText(
                        content,
                        getString(R.string.license_pl, siteUrl, siteUrl)
                )
            }
            ru.setOnClickListener {
                setContentText(
                        content,
                        getString(R.string.license_ru, siteUrl, siteUrl)
                )
            }
            de.setOnClickListener {
                setContentText(
                        content,
                        getString(R.string.license_de, siteUrl, siteUrl)
                )
            }
            fr.setOnClickListener {
                setContentText(
                        content,
                        getString(R.string.license_fr, siteUrl, siteUrl)
                )
            }
            es.setOnClickListener {
                setContentText(
                        content,
                        getString(R.string.license_es, siteUrl, siteUrl)
                )
            }
            it.setOnClickListener {
                setContentText(
                        content,
                        getString(R.string.license_it, siteUrl, siteUrl)
                )
            }
            pt.setOnClickListener {
                setContentText(
                        content,
                        getString(R.string.license_pt, siteUrl, siteUrl)
                )
            }
            ch.setOnClickListener {
                setContentText(
                        content,
                        getString(R.string.license_ch, siteUrl, siteUrl)
                )
            }
        }

        return dialog
    }

    companion object {

        val TAG: String = CC3LicenseDialogFragment::class.java.simpleName

        fun newInstance() = CC3LicenseDialogFragment()

        private fun setContentText(content: TextView, text: String) {
            content.text = Html.fromHtml(text)
        }
    }
}
